File: src/main/java/leetcode/lists/topinterview/LeetCode172.java
package leetcode.lists.topinterview;

/**
 * @author wujingxinit@outlook.com
 * @date 2022/9/27 13:14
 */
public class LeetCode172 {


    //https://www.youtube.com/watch?v=A5973V68Tvo
    public int trailingZeroes(int n) {
        int ans = 0;
        while (n != 0) {
            n /= 5;
            ans += n;
        }
        return ans;
    }
}

File: src/main/go/leetcode/lists/topinterview/LeetCode172.go
package topinterview

func trailingZeroes(n int) int {
	if n == 0 {
		return 0
	}
	// 以5的数量为准
	ans := 0
	for n != 0 {
		n /= 5
		ans += n
	}
	return ans
}
